pkg/controllers/routereflector/datastores/etcd.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 IBM Corporation All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastores
```

Comment:
Is there anything etcdv3 specific about the logic in this file other than the GetType() method?

Reply:
Can be simplified if we store datastore type in `type ctrl struct {` instead of the datasource struct. What is your advice?